cli/tests/unit/fetch_test.ts

```diff
@@ -1887,3 +1887,27 @@ Deno.test(
     await server;
   },
 );
+
+Deno.test(
+  { permissions: { net: true } },
+  async function fetchRequestBodyEmptyStream() {
```

Comment:
```
fetchRequestBodyEmptyStream => ./cli/tests/unit/fetch_test.ts:1891:6
error: AssertionError: Expected function to reject.
    throw new AssertionError(
          ^
    at assertRejects (file:///home/runner/work/deno/deno/test_util/std/testing/asserts.ts:891:11)
    at async fetchRequestBodyEmptyStream (file:///home/runner/work/deno/deno/cli/tests/unit/fetch_test.ts:1901:5)
```

Seems flaky https://github.com/denoland/deno/actions/runs/3904649696/jobs/6670704695

Reply:
i tried this test so many times locally and never was... why ci...

Reply:
I managed to reproduce it by running all the fetch tests together over and over again in two separate terminals. It didn't seem to occur with just one. `do { cargo run --bin deno -- test --allow-all --unstable --location=http://js-unit-tests/foo/bar .\cli\tests\unit\fetch_test.ts } until ($LastExitCode -ne 0)`

Reply:
hm, unsure what to do here

Reply:
I pushed a change to use a synchronous AbortSignal triggering to avoid the flakiness. I think this should be good since now it's not possible for the fetch to get resolved before the abort happens.